test/inductor/test_flex_flash.py

```diff
@@ -610,6 +707,47 @@ def test_flash_attention_mask_mod_cases(self, device, dtype, case):
         q, k, v = create_test_tensors(
             batch_size=case.batch_size,
             num_heads=case.num_heads,
+            num_heads_kv=case.num_heads_kv,
+            seq_len=case.seq_len,
+            dim=case.dim,
+            dtype=dtype,
+            device=device,
+            requires_grad=case.requires_grad,
+        )
+        flash_vs_triton(
+            q,
+            k,
+            v,
+            score_mod=(
+                case.score_mod_factory(dtype, device)
+                if case.score_mod_factory
+                else None
+            ),
+            block_mask=_create_block_mask_for_device(
+                case.mask_mod_factory(dtype, device),
+                case.batch_size,
+                case.block_mask_num_heads or case.num_heads,
+                case.seq_len,
+                case.seq_len,
+                device=device,
+            ),
+        )
+
+    @unittest.skip(
+        "GQA/MQA + block_mask produces incorrect results - needs investigation"
```

Comment:
skipping for now but the error is quite small and will be a lil more annoying to debug I imagine.

Reply:
Should it skip or xfail?

Reply:
We are good, I fixed here: https://github.com/Dao-AILab/flash-attention/pull/2098

was incorrectly doing the block_m reindex when kv_heads is > 1 but not pack-gqa set